### PR TITLE
Return ElementHandle from evaluate

### PR DESCRIPTION
**Pull request by JoelEinbinder**
If evaluation results in an Element, then an ElementHandle is returned. This is part 1 of #302.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -354,7 +354,7 @@ List of all available devices is available in the source code: [DeviceDescriptor
 - `...args` Arguments to pass to `pageFunction`
 - returns: <[Promise]> Resolves to the return value of `pageFunction`
 
-If the function, passed to the `page.evaluate`, returns a [Promise], then `page.evaluate` would wait for the promise to resolve and return it's value.
+If the function, passed to the `page.evaluate`, returns a [Promise], then `page.evaluate` would wait for the promise to resolve and return it's value. If the function returns an [Element], then `page.evaluate` will return an [ElementHandle].
```
Comment (ebidel) [inline docs/api.md]: return it's value -> return its value

File: docs/api.md [hunk 2]
```diff
@@ -969,7 +969,7 @@ Adds a `<script>` tag to the frame with the desired url. Alternatively, JavaScri
 - `...args` <...[string]> Arguments to pass to  `pageFunction`
 - returns: <[Promise]<[Object]>> Promise which resolves to function return value
 
-If the function, passed to the `page.evaluate`, returns a [Promise], then `page.evaluate` would wait for the promise to resolve and return it's value.
+If the function, passed to the `page.evaluate`, returns a [Promise], then `frame.evaluate` would wait for the promise to resolve and return it's value. If the function returns an [Element], then `frame.evaluate` will return an [ElementHandle].
```
Comment (ebidel) [inline docs/api.md]: return it's value -> return its value

File: docs/api.md [hunk 3]
```diff
@@ -1121,7 +1121,8 @@ The `elementHandle.dispose` method stops referencing the element handle.
 - `...args` <...[string]> Arguments to pass to  `pageFunction`
 - returns: <[Promise]<[Object]>> Promise which resolves to function return value
 
-If the function, passed to the `elementHandle.evaluate`, returns a [Promise], then `elementHandle.evaluate` would wait for the promise to resolve and return it's value.
+If the function, passed to the `elementHandle.evaluate`, returns a [Promise], then `elementHandle.evaluate` would wait for the promise to resolve and return it's value. If the function returns an [Element], then `elementHandle.evaluate` will return an [ElementHandle].
```
Comment (ebidel) [inline docs/api.md]: return it's value -> return its value

**Review by ebidel — COMMENTED** [review]
So fast!

**Comment by JoelEinbinder** [comment]
Build failed because of #351

**Comment by pavelfeldman** [comment]
Not sure how good an idea that is - we might want to expose object handles in a more generic way instead.

**Comment by aslushnikov** [comment]
Closing in favor of #382

**Pull request closed without merging**